packages/nimble-components/build/generate-workers/source/matrix-renderer.ts

```diff
@@ -8,16 +9,128 @@ import { expose } from 'comlink';
  * This setup is used in the wafer-map component to perform heavy computational duties
  */
 export class MatrixRenderer {
-    public dieMatrix: Uint8Array = Uint8Array.from([]);
+    public colIndexes: Uint32Array = Uint32Array.from([]);
+    public rowIndexes: Uint32Array = Uint32Array.from([]);
+    public canvas!: OffscreenCanvas;
+    public context!: OffscreenCanvasRenderingContext2D;
+    public values = new Float64Array([]);
+    public scaledColIndex = new Float64Array([]);
+    public scaledRowIndex = new Float64Array([]);
+    private scaleX: number = 1;
+    private scaleY: number = 1;
+    private baseX: number = 1;
+    private baseY: number = 1;
+    private dieDimensions: Dimensions = { width: 1, height: 1 };
+    private transform: Transform = { k: 1, x: 0, y: 0 };
+    private topLeftCanvasCorner: { x: number, y: number } = { x: 0, y: 0 };
+    private bottomRightCanvasCorner: { x: number, y: number } = { x: 500, y: 500 };
```

Comment:
Where do these initial values come from?

Reply:
I left them there to avoid the undefined kind of errors. I deleted the default values and put ! at the end of the properties, not sure if this is a proper solution